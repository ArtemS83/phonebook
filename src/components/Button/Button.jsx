import PropTypes from 'prop-types';
import style from './Button.module.scss';

const Button = ({ title, type, onClick }) => {
  return (
    <button
      data-testid="onClickFn"
      className={type === 'button' ? style.buttonButton : style.button}
      type={type}
      onClick={onClick}
    >
      {title}
    </button>
  );
};

Button.defaultProps = {
  title: 'Click me',
  type: 'button',
  onClick: () => {},
};

Button.propTypes = {
  title: PropTypes.string,
  type: PropTypes.string,
  onClick: PropTypes.func,
};

export default Button;
