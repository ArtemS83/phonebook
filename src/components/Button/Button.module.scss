.button {
  padding: 10px 20px;
  max-width: 130px;
  min-width: 60px;
  background-color: rgba(152, 226, 226, 0.74);
  // background-color: rgba(29, 73, 77, 0.683);
  border: 1px solid rgb(125, 19, 196);
  border-radius: 6px;
  box-shadow: 2px 6px 8px -2px #a09c9c;
  border-spacing: 0;
  outline: none;
  cursor: pointer;
  transition: all 300ms ease-in-out;
  &:hover,
  &:focus {
    background-color: #78beec; //78beec //rgb(206, 110, 211)
    transform: scale(1.03);
  }
}

.buttonButton {
  @extend .button;
  padding: 5px 10px;
  background-image: linear-gradient(
    120deg,
    rgba(246, 248, 248, 0.6) 0%,
    rgba(74, 75, 77, 0.6) 100%
  );
}
