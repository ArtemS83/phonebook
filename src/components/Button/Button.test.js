import { render } from '@testing-library/react';
import Button from './Button';

describe('Button', () => {
  it('Button should render properly without props', () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });

  it('Button should render properly type, title, onClick', () => {
    const testClick = jest.fn();
    const { container } = render(
      <Button title="Hello" type="submit" onClick={testClick} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('Button  should call  onClick-buttonRef', () => {
    const testClick = jest.fn();

    const { container } = render(
      <Button title="Hello click" onClick={testClick} />,
    );
    //типа костыль, если не знаем как определить элемент
    const buttonRef = container.querySelector('button');
    buttonRef.click();
    expect(testClick).toBeCalled();
  });

  it('Button  should call  onClick data-testId', () => {
    const testClick = jest.fn();

    const { getByTestId } = render(<Button onClick={testClick} />);

    getByTestId('onClickFn').click();

    expect(testClick).toBeCalled();
  });

  it('The default title should be Click me', () => {
    const { getByTestId } = render(<Button />);

    const title = getByTestId('onClickFn').textContent;

    expect(title).toBe('Click me');
  });

  it('title should be Add', () => {
    const { getByTestId } = render(<Button title="Add" />);

    const title = getByTestId('onClickFn').textContent;

    expect(title).toBe('Add');
  });
});
