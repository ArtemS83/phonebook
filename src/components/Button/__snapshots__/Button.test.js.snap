// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button should render properly type, title, onClick 1`] = `
<div>
  <button
    class="button"
    data-testid="onClickFn"
    type="submit"
  >
    Hello
  </button>
</div>
`;

exports[`Button Button should render properly without props 1`] = `
<div>
  <button
    class="buttonButton"
    data-testid="onClickFn"
    type="button"
  >
    Click me
  </button>
</div>
`;
